File: zul/src/org/zkoss/zul/api/Treefoot.java
/* Treefoot.java

{{IS_NOTE
	Purpose:
		
	Description:
		
	History:
		Tue Oct 22 14:45:31     2008, Created by Flyworld
}}IS_NOTE

Copyright (C) 2008 Potix Corporation. All Rights Reserved.

{{IS_RIGHT
	This program is distributed under GPL Version 3.0 in the hope that
	it will be useful, but WITHOUT ANY WARRANTY.
}}IS_RIGHT
 */
package org.zkoss.zul.api;

/**
 * A row of {@link Treefooter}.
 * 
 * <p>
 * Like {@link Treecols}, each tree has at most one {@link Treefoot}.
 * <p>
 * Default {@link #getZclass}: z-treefoot (since 5.0.0)
 * 
 * @author tomyeh
 * @since 3.5.2
 */
public interface Treefoot extends org.zkoss.zul.impl.api.XulElement {
	/**
	 * Returns the tree that it belongs to.
	 * <p>
	 * It is the same as {@link #getParent}.
	 */
	public org.zkoss.zul.api.Tree getTreeApi();

}

File: zul/src/main/resources/web/js/zul/sel/Treefoot.ts
/* Treefoot.ts

	Purpose:

	Description:

	History:
		Wed Jun 10 15:32:41     2009, Created by jumperchen

Copyright (C) 2009 Potix Corporation. All Rights Reserved.

This program is distributed under LGPL Version 2.0 in the hope that
it will be useful, but WITHOUT ANY WARRANTY.
*/
/**
 * A row of {@link Treefooter}.
 *
 * <p>Like {@link Treecols}, each tree has at most one {@link Treefoot}.
 * @defaultValue {@link getZclass}: z-treefoot
 */
@zk.WrapClass('zul.sel.Treefoot')
export class Treefoot extends zul.Widget {
	override parent!: zul.sel.Tree | undefined;
	/**
	 * @returns the tree that it belongs to.
	 */
	getTree(): zul.sel.Tree | undefined {
		return this.parent;
	}

	//bug #3014664
	override setVflex(vflex?: boolean | string): this { //vflex ignored for Treefoot
		vflex = false;
		return super.setVflex(vflex);
	}

	//bug #3014664
	override setHflex(hflex: boolean | string): this { //hflex ignored for Treefoot
		hflex = false;
		return super.setHflex(hflex);
	}

	/** @internal */
	override deferRedrawHTML_(out: string[]): void {
		out.push('<tr', this.domAttrs_({domClass: true}), ' class="z-renderdefer"></tr>');
	}
}